cmd/conduit/root/pipelines/init.go

```diff
@@ -0,0 +1,253 @@
+	}
+}
+
+func (c *InitCommand) getSourceSpec() (connectorSpec, error) {
+	for _, conn := range builtin.DefaultBuiltinConnectors {
+		specs := conn.NewSpecification()
+		if specs.Name == c.flags.Source || specs.Name == "builtin:"+c.flags.Source {
+			if conn.NewSource == nil {
+				return connectorSpec{}, cerrors.Errorf("plugin %v has no source", c.flags.Source)
+			}
+
+			return connectorSpec{
+				Name:   specs.Name,
+				Params: conn.NewSource().Parameters(),
+			}, nil
+		}
+	}
+
+	return connectorSpec{}, cerrors.Errorf("%v: %w", c.flags.Source, plugin.ErrPluginNotFound)
+}
+
+func (c *InitCommand) getDestinationSpec() (connectorSpec, error) {
+	for _, conn := range builtin.DefaultBuiltinConnectors {
+		specs := conn.NewSpecification()
+		if specs.Name == c.flags.Destination || specs.Name == "builtin:"+c.flags.Destination {
+			if conn.NewDestination == nil {
+				return connectorSpec{}, cerrors.Errorf("plugin %v has no source", c.flags.Destination)
+			}
+
+			return connectorSpec{
+				Name:   specs.Name,
+				Params: conn.NewDestination().Parameters(),
+			}, nil
+		}
+	}
+	return connectorSpec{}, cerrors.Errorf("%v: %w", c.flags.Destination, plugin.ErrPluginNotFound)
+}
+
+// getDemoSourceGeneratorSpec returns a simplified version of the source generator connector.
+func (c *InitCommand) getDemoSourceGeneratorSpec(spec connectorSpec) connectorSpec {
+	return connectorSpec{
+		Name: defaultSource,
+		Params: map[string]config.Parameter{
+			"format.type": {
+				Description: spec.Params["format.type"].Description,
+				Type:        spec.Params["format.type"].Type,
+				Default:     "structured",
+				Validations: spec.Params["format.type"].Validations,
+			},
+			"format.options.scheduledDeparture": {
+				Description: "Generate field 'scheduledDeparture' of type 'time'",
+				Type:        config.ParameterTypeString,
+				Default:     "time",
+			},
+			"format.options.airline": {
+				Description: "Generate field 'airline' of type string",
+				Type:        config.ParameterTypeString,
+				Default:     "string",
+			},
+			"rate": {
+				Description: spec.Params["rate"].Description,
+				Type:        spec.Params["rate"].Type,
+				Default:     "1",
+			},
+		},
+	}
+}
+
+// getDemoDestinationFileSpec returns a simplified version of the destination file connector.
+func (c *InitCommand) getDemoDestinationFileSpec(spec connectorSpec) connectorSpec {
+	return connectorSpec{
+		Name: defaultDestination,
+		Params: map[string]config.Parameter{
+			"path": {
+				Description: spec.Params["path"].Description,
+				Type:        spec.Params["path"].Type,
+				Default:     "./destination.txt",
+			},
+		},
+	}
+}
+
+func (c *InitCommand) buildTemplatePipeline() (pipelineTemplate, error) {
+	srcSpec, err := c.getSourceSpec()
+	if err != nil {
+		return pipelineTemplate{}, cerrors.Errorf("failed getting source params: %w", err)
+	}
+
+	// provide a simplified version
+	if c.flags.Source == defaultSource {
```

Comment:
I'm thinking, what if someone would like to configure a `generator` source himself?